agents-for-bedrock/features-examples/11-create-agent-with-code-interpreter/11-create-agent-with-code-interpreter.ipynb

```diff
@@ -0,0 +1,2479 @@
+   "metadata": {},
+   "source": [
+    "Next we want to import the support packages and set the logger object"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "5255655e-d83e-416a-a459-83590483d22b",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import json\n",
+    "import time\n",
+    "import zipfile\n",
+    "from io import BytesIO\n",
+    "import uuid\n",
+    "import pprint\n",
+    "import logging"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "593be615-dab2-4084-a518-ee905113bfc7",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# setting logger\n",
+    "logging.basicConfig(format='[%(asctime)s] p%(process)s {%(filename)s:%(lineno)d} %(levelname)s - %(message)s', level=logging.INFO)\n",
+    "logger = logging.getLogger(__name__)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "52dad4b8-c2e0-44cf-99ee-5c2a7e1a600f",
+   "metadata": {},
+   "source": [
+    "Let's now create the boto3 clients for the required AWS services"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "67883027-bbb5-42a6-955d-0aa019a8e4b3",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "[2024-07-14 12:07:30,476] p15000 {credentials.py:1278} INFO - Found credentials in shared credentials file: ~/.aws/credentials\n"
+     ]
+    }
+   ],
+   "source": [
+    "# getting boto3 clients for required AWS services\n",
+    "sts_client = boto3.client('sts')\n",
+    "iam_client = boto3.client('iam')\n",
+    "lambda_client = boto3.client('lambda')\n",
+    "bedrock_agent_client = boto3.client('bedrock-agent')\n",
+    "bedrock_agent_runtime_client = boto3.client('bedrock-agent-runtime')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "71505c3b-c2a5-47dc-a427-77f21ab50f0d",
+   "metadata": {},
+   "source": [
+    "Next we can set some configuration variables for the agent and for the lambda function being created"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "f6af5cd4-33b8-40c3-9d99-1790540eb60c",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "('us-east-1', '271976243307')"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "session = boto3.session.Session()\n",
+    "region = session.region_name\n",
+    "account_id = sts_client.get_caller_identity()[\"Account\"]\n",
+    "region, account_id"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "id": "0dcb5332-c035-435a-91e5-838ec95d9a7d",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "# configuration variables\n",
+    "suffix = f\"{region}-{account_id}\"\n",
+    "agent_name = \"assistant-w-code-interpret\"\n",
```

Comment:
assistant-w-code-interpretation --- we need to use the marketing name

Reply:
Ah, it's too long, that's why I shortened it...